src/protocol/libp2p/kademlia/mod.rs

```diff
@@ -738,17 +738,36 @@ impl Kademlia {
                                 self.routing_table.closest(Key::from(peer), self.replication_factor).into()
                             );
                         }
-                        Some(KademliaCommand::PutRecord { record, query_id }) => {
+                        Some(KademliaCommand::PutRecord { record, query_id, peers }) => {
                             tracing::debug!(target: LOG_TARGET, ?query_id, key = ?record.key, "store record to DHT");
 
-                            self.store.put(record.clone());
                             let key = Key::new(record.key.clone());
 
-                            self.engine.start_put_record(
-                                query_id,
-                                record,
-                                self.routing_table.closest(key, self.replication_factor).into(),
-                            );
+
+                            if let Some(peers) = peers {
+                                // Put the record to the specified peers.
+                                let peers: VecDeque<_> = peers.into_iter().filter_map(|peer| {
+                                    if let KBucketEntry::Occupied(entry) = self.routing_table.entry(Key::from(peer)) {
+                                        Some(entry.clone())
+                                    } else {
+                                        None
+                                    }
+                                }).collect();
+
+                                self.engine.start_put_record(
+                                    query_id,
+                                    record,
+                                    peers
+                                );
```

Comment:
Going once again through `start_put_record()` implementation and [`FindNodeContext`](https://github.com/paritytech/litep2p/blob/d7fa12be82423393519644171b43a009b6a3938c/src/protocol/libp2p/kademlia/query/find_node.rs) query used under the hood, this seems to be doing a slightly different thing that what's needed by authority-discovery. Seeding `peers` with some specific candidates would result in `QueryType::PutRecord` query initialized by that peers, but if `FindNode` collects more than 20 peers that are closer to the target than the original "seed" peers, the "seed" peers would end up not updated.

To update specific peers, we can go two routes:
1. Update only peers if they are already in the routing table. For this, we just need to send [`KademliaMessage::put_value`](https://github.com/paritytech/litep2p/blob/d7fa12be82423393519644171b43a009b6a3938c/src/protocol/libp2p/kademlia/message.rs#L82) directly to peers from the routing table. This should be enough for authority-discivery needs, as the peers should be already in the routing table.
2. Update peers, even if they are not in the routing table. Basically, do `FindNode` query for the needed peers first, followed by sending `KademliaMessage::put_value` to the discovered addresses. We likely need this to conform to libp2p `put_record_to` implementation, but can do in a follow-up.